internal/lock/lockfile.go

```diff
@@ -24,24 +23,41 @@ type File struct {
 	devboxProject
 	resolver
 
-	LockFileVersion string              `json:"lockfile_version"`
-	Packages        map[string]*Package `json:"packages"`
+	LockFileVersion string `json:"lockfile_version"`
+
+	// Packages is keyed by "canonicalName@version"
```

Comment:
This is not strictly true. packages are keyed by any value that can go in `packages` in devbox.json

Reply:
isn't that always with `@latest` or `@<concrete-version>` ?